src/ol_infrastructure/applications/botkube/__main__.py

```diff
@@ -0,0 +1,673 @@
+    vault_auth_endpoint=cluster_stack.require_output("vault_auth_endpoint"),
+    vault_auth_role_name=botkube_vault_auth_backend_role.role_name,
+)
+
+vault_k8s_resources = OLVaultK8SResources(
+    resource_config=vault_k8s_resources_config,
+    opts=ResourceOptions(
+        delete_before_replace=True,
+        depends_on=[botkube_vault_auth_backend_role],
+    ),
+)
+
+# Load the static secrets into a k8s secret via VSO
+static_secrets_name = "communication-slack"  # pragma: allowlist secret
+static_secrets = OLVaultK8SSecret(
+    name=f"botkube-{stack_info.env_suffix}-static-secrets",
+    resource_config=OLVaultK8SStaticSecretConfig(
+        name="botkube-static-secrets",
+        namespace=botkube_namespace,
+        labels=k8s_global_labels,
+        dest_secret_name=static_secrets_name,
+        dest_secret_labels=k8s_global_labels,
+        mount="secret-botkube",
+        mount_type="kv-v2",
+        path="secrets",
+        includes=["*"],
+        excludes=[],
+        exclude_raw=True,
+        vaultauth=vault_k8s_resources.auth_name,
+    ),
+    opts=ResourceOptions(
+        delete_before_replace=True,
+        parent=vault_k8s_resources,
+        depends_on=[botkube_static_vault_secrets],
+    ),
+)
+# end Vault hoo-ha
+
+
+setup_k8s_provider(kubeconfig=cluster_stack.require_output("kube_config"))
+
+cluster_stack.require_output("namespaces").apply(
+    lambda ns: check_cluster_namespace(botkube_namespace, ns)
+)
+
+# Install the botkube helm chart
+botkube_application = kubernetes.helm.v3.Release(
+    f"botkube-{stack_info.name}-application-helm-release",
+    kubernetes.helm.v3.ReleaseArgs(
+        name="botkube",
+        chart="botkube",
+        version=BOTKUBE_CHART_VERSION,
+        namespace=botkube_namespace,
+        cleanup_on_fail=True,
+        repository_opts=kubernetes.helm.v3.RepositoryOptsArgs(
+            repo="https://charts.botkube.io",
+        ),
+        values={
+            "commonLabels": k8s_global_labels,
+            # Botkube Helm values converted to Python dictionary
+            # Botkube image configuration
+            "image": {
+                "registry": "ghcr.io",
+                "repository": "kubeshop/botkube",
+                "pullPolicy": "IfNotPresent",
+                "tag": "v1.14.0",
+            },
+            # Pod Security Policy configuration
+            "podSecurityPolicy": {"enabled": False},
+            # Security context configuration
+            "securityContext": {"runAsUser": 101, "runAsGroup": 101},
+            # Container security context
+            "containerSecurityContext": {
+                "privileged": False,
+                "allowPrivilegeEscalation": False,
+                "readOnlyRootFilesystem": True,
+            },
+            # RBAC configuration
+            "rbac": {
+                "serviceAccountMountPath": "/var/run/7e7fd2f5-b15d-4803-bc52-f54fba357e76/secrets/kubernetes.io/serviceaccount",  # noqa: E501
```

Comment:
This seems like it's unnecessary to specify explicitly?

Reply:
Looking at the docs, this RBAC thing seems to be the path that plugins use to access the k8s APIs safely.

I'd need to understand the details of that more than I do to be willing to change this.